### ISSUE TITLE
Fails while rebasing

### ISSUE THREAD

**Issue by Spunkie** [kevinoid/git-branch-is#58]
I'm using `git-branch-is` in a `husky` pre-commit hook to stop commits to the master branch. It works well enough when doing normal commits but when trying to do an interactive rebase it seems  `git-branch-is`  errors out.

Here's a snippet of my `package.json` and a screenshot of the error
```
{
	...
	"scripts": {
		...
		"no-master-commit": "git-branch-is --not master"
	},
	"lint-staged": {
		"*.{js,vue,html}": "eslint --fix"
	},
	"husky": {
		"hooks": {
			"pre-commit": "npm run no-master-commit && lint-staged"
		}
	},
	...
}
```
![TortoiseGitProc_J9qOqV6ZEJA3tjI4JdUr17N1QkX8G8X6](https://user-images.githubusercontent.com/1653976/88997723-90477a00-d2ad-11ea-9c7e-daf81101bfb1.png)

**Comment by Spunkie** [comment]
And here's `debug.log` mentioned in the error message:
```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\_GIT\\monday.com-cro-solution\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'no-master-commit'
1 verbose cli ]
2 info using npm@6.14.7
3 info using node@v12.10.0
4 verbose run-script [ 'preno-master-commit', 'no-master-commit', 'postno-master-commit' ]
5 info lifecycle monday.com-cro-solution@1.1.0~preno-master-commit: monday.com-cro-solution@1.1.0
6 info lifecycle monday.com-cro-solution@1.1.0~no-master-commit: monday.com-cro-solution@1.1.0
7 verbose lifecycle monday.com-cro-solution@1.1.0~no-master-commit: unsafe-perm in lifecycle true
8 verbose lifecycle monday.com-cro-solution@1.1.0~no-master-commit: PATH: C:\_GIT\monday.com-cro-solution\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\_GIT\monday.com-cro-solution\node_modules\.bin;C:\Users\SPUNKI~1\AppData\Local\Temp\yarn--1596166251076-0.6172443262497098;C:\_GIT\monday.com-cro-solution\node_modules\.bin;C:\Users\Spunkie-NEW\AppData\Local\Yarn\Data\link\node_modules\.bin;C:\Program Files\libexec\lib\node_modules\npm\bin\node-gyp-bin;C:\Program Files\lib\node_modules\npm\bin\node-gyp-bin;C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;C:\Program Files\Git\mingw64\libexec\git-core;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Users\Spunkie-NEW\bin;C:\tools\ruby27\bin;C:\Python27;C:\Python27\Scripts;C:\Program Files\AdoptOpenJDK\jdk-11.0.6.10-hotspot\bin;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Python38\Scripts;C:\Python38;C:\tools\ruby26\bin;C:\Python37\Scripts;C:\Python37;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files\NVIDIA Corporation\NVIDIA NGX;C:\ProgramData\chocolatey\bin;C:\tools\php73;C:\ProgramData\ComposerSetup\bin;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\OpenSSL\bin;C:\Program Files\Microsoft VS Code\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\nodejs;C:\Program Files\dotnet;C:\tools\php74;C:\Program Files\TortoiseGit\bin;C:\Program Files (x86)\Yarn\bin;C:\Program Files\Git\cmd;C:\HashiCorp\Vagrant\bin;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Go\bin;C:\Program Files (x86)\GnuPG\bin;C:\Users\Spunkie-NEW\AppData\Local\Microsoft\WindowsApps;C:\Users\Spunkie-NEW\AppData\Roaming\Composer\vendor\bin;C:\Users\Spunkie-NEW\go\bin;C:\Program Files\Oracle\VirtualBox;C:\Users\Spunkie-NEW\AppData\Roaming\npm;C:\Users\Spunkie-NEW\AppData\Local\Yarn\bin;C:\Program Files\heroku\bin;C:\Users\Spunkie-NEW\AppData\Local\Markdown Monster;C:\Users\Spunkie-NEW\go\bin;C:\Program Files (x86)\EaseUS\Todo Backup\bin\x64;C:\Program Files\Git\usr\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\mingw64\libexec\git-core
9 verbose lifecycle monday.com-cro-solution@1.1.0~no-master-commit: CWD: C:\_GIT\monday.com-cro-solution
10 silly lifecycle monday.com-cro-solution@1.1.0~no-master-commit: Args: [ '/d /s /c', 'git-branch-is --not master' ]
11 silly lifecycle monday.com-cro-solution@1.1.0~no-master-commit: Returned: code: 128  signal: null
12 info lifecycle monday.com-cro-solution@1.1.0~no-master-commit: Failed to exec no-master-commit script
13 verbose stack Error: monday.com-cro-solution@1.1.0 no-master-commit: `git-branch-is --not master`
13 verbose stack Exit status 128
13 verbose stack     at EventEmitter.<anonymous> (C:\_GIT\monday.com-cro-solution\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:209:13)
13 verbose stack     at ChildProcess.<anonymous> (C:\_GIT\monday.com-cro-solution\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:209:13)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
14 verbose pkgid monday.com-cro-solution@1.1.0
15 verbose cwd C:\_GIT\monday.com-cro-solution
16 verbose Windows_NT 10.0.17763
17 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\_GIT\\monday.com-cro-solution\\node_modules\\npm\\bin\\npm-cli.js" "run" "no-master-commit"
18 verbose node v12.10.0
19 verbose npm  v6.14.7
20 error code ELIFECYCLE
21 error errno 128
22 error monday.com-cro-solution@1.1.0 no-master-commit: `git-branch-is --not master`
22 error Exit status 128
23 error Failed at the monday.com-cro-solution@1.1.0 no-master-commit script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 128, true ]
```

**Comment by kevinoid** [comment]
Good find @Spunkie, that looks like a bug to me.  Minimal repro:
```sh
GIT_SEQUENCE_EDITOR="sed -i -re 's/^pick /edit /'" git rebase -i HEAD^
git-branch-is master
```
Which prints

    Error: Command failed: git symbolic-ref --short HEAD
    fatal: ref HEAD is not a symbolic ref

Note that the state is not on any branch, but there is information to find which branch will be checked out when rebase finishes.  e.g. `git branch` prints:

    * (no branch, rebasing master)
    master

For your use case, are you expecting `git-branch-is --not master` to be true or false when rebasing `master`?

**Comment by Spunkie** [comment]
> For your use case, are you expecting `git-branch-is --not master` to be true or false when rebasing master?

In this specific use case I would definitely expect `git-branch-is --not master` to return `true`. Since from my perspective I'm just updating the `pr_tester` branch to the latest and don't expect anything in `master` to actually change?

**Comment by kevinoid** [comment]
Thanks @Spunkie!  I just pushed a few commits which should fix the issue.  If you have a moment, could you test with `master` to see if it behaves the way you'd expect:
```sh
npm install 'git-branch-is@git+https://github.com/kevinoid/git-branch-is.git#master'
```
If no issues come up, I'll cut a release shortly.

**Comment by Spunkie** [comment]
I took 2 quick passes at testing, in both I ran 
```
yarn remove git-branch-is
yarn add -D 'git-branch-is@git+https://github.com/kevinoid/git-branch-is.git#master'
```

1. The exact same setup as reported, ran a rebase, and works fine now. 🎉
2. I switched back to master and tried to commit, this is still rejecting as expected. 👍

I didn't get a chance to test what happens if you actually try to rebase `master` though.
___
I appreciate the ⚡ quick turnaround on this, thanks @kevinoid

**Comment by kevinoid** [comment]
Thanks @Spunkie!  Much appreciated!  Released as [v4.0.0](https://github.com/kevinoid/git-branch-is/releases/tag/v4.0.0).

For reference: Currently `git-branch-is master` will fail (and `git-branch-is --not master` succeed) when rebasing branch `master`.  There may be a use case for detecting the branch of a rebase-in-progress, but I figured we can cross that bridge when we come to it.
